packages/implementations-lib/src/transfers/mongo_transfers_repo.ts

```diff
@@ -97,153 +97,175 @@ export class MongoTransfersRepo implements ITransfersRepository {
-            filter.$and.push({updatedAt: {$gte:startDate}});
-        }
-        if(endDate){
-            filter.$and.push({updatedAt: {$lte:endDate}});
-        }
-        if(filter.$and.length === 0) {
-            filter = {};
-        }
-
-        try {
-            const skip = Math.floor(pageIndex * pageSize);
-            const result = await this.transfers.find(
-                filter,
-                {
-                    sort:["updatedAt", "desc"],
-                    projection: {_id: 0},
-                    skip: skip,
-                    limit: pageSize
-                }
-            ).toArray().catch((e: unknown) => {
-                this._logger.error(`Unable to get transfers: ${(e as Error).message}`);
-                throw new UnableToGetTransferError();
-            });
-
-            const countResult = await this.transfers.countDocuments(filter).catch(() => {
-                this._logger.error("Unable to get transfers count");
-            }) || result.length;
-
-            searchResults.items = result.map(this._mapToTransfer);
-
-            searchResults.totalPages = Math.ceil(countResult / pageSize);
-            searchResults.pageSize = Math.max(pageSize, result.length);
-
-        } catch (err) {
-            this._logger.error(err);
-        }
-
-        return Promise.resolve(searchResults);
+		state: string | null,
+		transferType: string | null,
+		payerIdType: string | null,
+		payeeIdType: string | null,
+		currency: string | null,
+		id: string | null,
+		payerIdValue: string | null,
+		payeeIdValue: string | null,
+		bulkTransferId: string | null,
+		startDate: number,
+		endDate: number,
+		pageIndex = 0,
+		pageSize: number = MAX_ENTRIES_PER_PAGE
+	): Promise<TransfersSearchResults> {
+		// make sure we don't go over or below the limits
+		pageSize = Math.min(pageSize, MAX_ENTRIES_PER_PAGE);
+		pageIndex = Math.max(pageIndex, 0);
+
+		const searchResults: TransfersSearchResults = {
+			pageSize: pageSize,
+			pageIndex: pageIndex,
+			totalPages: 0,
+			items: []
+		};
+
+		let filter: any = { $and: [] }; // eslint-disable-line @typescript-eslint/no-explicit-any
+
+		if (id) {
+			filter.$and.push({ "transferId": { "$regex": id, "$options": "i" } });
+		}
+		if (state) {
+			filter.$and.push({ transferState: state });
+		}
+		if (currency) {
+			filter.$and.push({ currencyCode: currency });
+		}
+		if (startDate) {
+			filter.$and.push({ updatedAt: { $gte: startDate } });
+		}
+		if (endDate) {
+			filter.$and.push({ updatedAt: { $lte: endDate } });
+		}
+
+		if (payerIdValue) {
+			filter.$and.push({ payerFspId: payerIdValue });
+		}
+
+		if (payeeIdValue) {
+			filter.$and.push({ payeeFspId: payeeIdValue });
+		}
+
+		if (transferType) {
+			filter.$and.push({ transferType: transferType });
+		}
+
+		if (payerIdType) {
+			filter.$and.push({ payerIdType: payerIdType });
+		}
+
+		if (payeeIdType) {
+			filter.$and.push({ payeeIdType: payeeIdType });
+		}
+
+		if (bulkTransferId) {
```

Comment:
Is it missing a regex here (like in line 153)? Since it's not on the search keywords

Reply:
> Is it missing a regex here (like in line 153)? Since it's not on the search keywords

Thanks, @ruirochawork . Updated the pull request. Please check.